editoast/src/views/infra/mod.rs

```diff
@@ -561,6 +565,79 @@ pub mod tests {
         assert!(Infra::delete(db_pool, cloned_infra_id).await.unwrap());
     }
 
+    #[derive(QueryableByName)]
+    struct Count {
+        #[diesel(sql_type = BigInt)]
+        nb: i64,
+    }
+
+    #[rstest]
+    async fn infra_clone(db_pool: Data<DbPool>) {
+        let app = create_test_service().await;
+        let small_infra = &small_infra(db_pool.clone()).await.model;
+        let small_infra_id = small_infra.id.unwrap();
+
+        let pg_config = PostgresConfig::default();
+        let pg_config_url = pg_config.url().expect("cannot get postgres config url");
+        let mut conn = PgConnection::establish(pg_config_url.as_str())
+            .await
+            .expect("Error while connecting DB");
+
+        let infra_cache = InfraCache::load(&mut conn, small_infra).await.unwrap();
+
+        generated_data::refresh_all(db_pool.clone(), small_infra_id, &infra_cache)
+            .await
+            .unwrap();
+
+        let errors_count =
+            sql_query("SELECT COUNT (*) as nb from infra_layer_error where infra_id = $1")
+                .bind::<BigInt, _>(small_infra_id)
+                .get_result::<Count>(&mut conn)
+                .await
+                .unwrap();
+        assert!(errors_count.nb > 1);
+
+        let req_clone = TestRequest::post()
+            .uri(format!("/infra/{}/clone/?name=cloned_infra", small_infra_id).as_str())
+            .to_request();
+        let response = call_service(&app, req_clone).await;
+        assert_eq!(response.status(), StatusCode::OK);
+
+        let cloned_infra_id: i64 = read_body_json(response).await;
+
+        for object in ObjectType::iter() {
+            let model_table = object.get_table();
+
+            let mut vec_table = vec![model_table];
+
+            if let Some(layer_table) = object.get_geometry_layer_table() {
+                vec_table.push(layer_table);
+            }
+
+            for table in vec_table {
+                let mut reference = 0;
+                for inf_id in [small_infra_id, cloned_infra_id] {
```

Comment:
Instead of using the `reference` variable, we can store counts in a hashmap like `HashMap<TableName, Vec<usize>>`. Once the hashmap is created we can iterate on it to check the consistency and also check that all values are greater than 1.

Reply:
Agree for the idee of the hashmap, but not the second part of check that all values are greater than 1 (see my comment above)